File: HNRZ Salary Manager/app/Models/PayrollMethod.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\SoftDeletes;

class PayrollMethod extends Model
{
    use HasFactory, SoftDeletes;

        protected $fillable = [
        'name',
        'type',
        'description',
        'is_active',
    ];

    protected function casts(): array
    {
        return [
            'is_active' => 'boolean',
        ];
    }

    public function employees()
    {
        return $this->hasMany(Employee::class, 'payroll_method_id');
    }
}

File: hnrz_salary_manager/lib/models/payroll_method.dart
class PayrollMethod {
  final int id;
  final String type;
  final String name;
  final String? description;
  final bool isActive;

  PayrollMethod({
    required this.id,
    required this.type,
    required this.name,
    this.description,
    required this.isActive,
  });

  factory PayrollMethod.fromJson(Map<String, dynamic> json) {
    return PayrollMethod(
      id: json['id'],
      type: json['type'],
      name: json['name'],
      description: json['description'],
      isActive: json['is_active'] == true || json['is_active'] == 1,
    );
  }
}